bin/distmap

```diff
@@ -20,124 +20,166 @@ use DownloadTrimmedRead;
 use DataCleanup;
 use ReadToolsUtils;
 
-my $script = "distmap";
+# current version
+use constant VERSION => "3.1.0-alpha";
 
-my $version = "3.1.0-alpha";
-
-### This number need to define. How many processors Hadoop can use on each nodes.
+## TODO - make a constant!!
+## TODO - should be exposed? or maybe depends on other variable (e.g., priority?)
+### This number need to define. How many processors Hadoop can use on each nodes
 my $number_of_processors = 3; #18;
 
 
+
+# ARGUMENT VARIABLES
+## REQUIRED
 my $reference_fasta="";
-my $refindex_archive="";
-my @input_files = ();
 my $output_directory="";
+my @input_files = ();
+my $mapper_args="";
+my @mapper=();
+my @mapper_path=();
+my @mapper_args=();
+
+## PSEUDO-REQUIRED
+### TODO: should this be called after parsing if not provided? I think that it would be more consistent.
```

Comment:
I like it when scripts set their default values at the top, because it easier to see what sort of values go into the variables and what the defaults are, without having to dig deeper into the script. I find it also slightly more parsimonious (saves a check and one assignment).

Reply:
Removed comment.